src/Libraries/Microsoft.Extensions.TimeProvider.Testing/FakeTimeProvider.cs

```diff
@@ -55,31 +61,26 @@ public override DateTimeOffset GetUtcNow()
     /// <param name="value">The date and time in the UTC timezone.</param>
     public void SetUtcNow(DateTimeOffset value)
     {
-        List<FakeTimeProviderTimer.Waiter> waiters;
-        lock (Waiters)
+        if (value < _now)
         {
-            _now = value;
-            waiters = GetWaitersToWake();
+            Throw.ArgumentOutOfRangeException(nameof(value), $"Cannot go back in time. Current time is {GetUtcNow()}.");
         }
 
-        WakeWaiters(waiters);
+        while (_now <= value && TryGetWaiterToWake(value) is FakeTimeProviderTimer.Waiter waiter)
```

Comment:
I'm finding it mentally challenging to parse `_now <= value` - `_now` is essentially a const in this context, and `if (10 <= value)` is harder to parse than `if (value >= 10)`.

```suggestion
        while (value >= _now && TryGetWaiterToWake(value) is FakeTimeProviderTimer.Waiter waiter)
```

Reply:
Fair enough, I have no strong opening here. 

In my head, `_now` is supposed to be before (we are moving forward in time), so I find it easier to read when it is first, i.e. the lower value comes before the larger value.